From 38035ea4fd2fb1561f6c832057641dcedc3f73db Mon Sep 17 00:00:00 2001
Message-Id: <38035ea4fd2fb1561f6c832057641dcedc3f73db.1425610844.git.chang-joon.lee@intel.com>
In-Reply-To: <6fa2ab23adf6d7de781e906d04a0224410262d9b.1425610844.git.chang-joon.lee@intel.com>
References: <6fa2ab23adf6d7de781e906d04a0224410262d9b.1425610844.git.chang-joon.lee@intel.com>
From: Deepak S <deepak.s@intel.com>
Date: Sat, 28 Feb 2015 11:30:38 +0530
Subject: [PATCH 27/27] FOR_UPSTREAM [VPG]: drm/i915: Clean-up idr table if
 context create fails.

Cleanup idr table if any error happens after __create_hw_context() in
i915_gem_create_context()

Issue: GMINL-6551
Change-Id: I0186efa71b0ada34b74e3a8f12297d4636a6ff38
Signed-off-by: Deepak S <deepak.s@intel.com>
---
 drivers/gpu/drm/i915/i915_gem_context.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/gpu/drm/i915/i915_gem_context.c b/drivers/gpu/drm/i915/i915_gem_context.c
index 75fca5f..38de584 100644
--- a/drivers/gpu/drm/i915/i915_gem_context.c
+++ b/drivers/gpu/drm/i915/i915_gem_context.c
@@ -322,6 +322,8 @@ err_unpin:
 	if (is_global_default_ctx && ctx->legacy_hw_ctx.rcs_state)
 		i915_gem_object_ggtt_unpin(ctx->legacy_hw_ctx.rcs_state);
 err_destroy:
+	if (ctx->file_priv)
+		idr_remove(&ctx->file_priv->context_idr, ctx->user_handle);
 	i915_gem_context_unreference(ctx);
 	return ERR_PTR(ret);
 }
-- 
1.7.9.5

